x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/ApiKeyFieldNameTranslators.java

```diff
@@ -38,37 +61,243 @@ public class ApiKeyFieldNameTranslators {
+            }
+            if (FieldSortBuilder.DOC_FIELD_NAME.equals(fieldSortBuilder.getFieldName())) {
+                searchSourceBuilder.sort(fieldSortBuilder);
+            } else {
+                final String translatedFieldName = translate(fieldSortBuilder.getFieldName());
+                fieldNameVisitor.accept(translatedFieldName);
+                if (translatedFieldName.equals(fieldSortBuilder.getFieldName())) {
+                    searchSourceBuilder.sort(fieldSortBuilder);
+                } else {
+                    final FieldSortBuilder translatedFieldSortBuilder = new FieldSortBuilder(translatedFieldName).order(
+                        fieldSortBuilder.order()
+                    )
+                        .missing(fieldSortBuilder.missing())
+                        .unmappedType(fieldSortBuilder.unmappedType())
+                        .setFormat(fieldSortBuilder.getFormat());
+
+                    if (fieldSortBuilder.sortMode() != null) {
+                        translatedFieldSortBuilder.sortMode(fieldSortBuilder.sortMode());
+                    }
+                    if (fieldSortBuilder.getNestedSort() != null) {
+                        translatedFieldSortBuilder.setNestedSort(fieldSortBuilder.getNestedSort());
+                    }
+                    if (fieldSortBuilder.getNumericType() != null) {
+                        translatedFieldSortBuilder.setNumericType(fieldSortBuilder.getNumericType());
+                    }
+                    searchSourceBuilder.sort(translatedFieldSortBuilder);
+                }
+            }
+        });
+    }
+
+    /**
+     * Deep copies the passed-in {@param queryBuilder} translating all the field names, from query level to index level,
+     * see {@link  #translate}. In general, the returned builder should create the same query as if the query were
+     * created by the passed in {@param queryBuilder}, only with the field names translated.
+     * Field name patterns (including "*"), are also replaced with the explicit index level field names whose
+     * associated query level field names match the pattern.
+     * The optional {@param visitor} can be used to collect all the translated field names.
+     */
+    public static QueryBuilder translateQueryBuilderFields(QueryBuilder queryBuilder, @Nullable Consumer<String> visitor) {
+        Objects.requireNonNull(queryBuilder, "unsupported \"null\" query builder for field name translation");
+        final Consumer<String> fieldNameVisitor = visitor != null ? visitor : ignored -> {};
+        if (queryBuilder instanceof final BoolQueryBuilder query) {
+            final BoolQueryBuilder newQuery = QueryBuilders.boolQuery()
+                .minimumShouldMatch(query.minimumShouldMatch())
+                .adjustPureNegative(query.adjustPureNegative())
+                .boost(query.boost())
+                .queryName(query.queryName());
```

Comment:
`queryName` was another parameter, like `boost` that's present for all query types, that we originally didn't carry through in the copy.